src/components/TODOList/TODOListMainView.js

```diff
@@ -0,0 +1,30 @@
+import React, { useState } from 'react';
+import TODOListViewer from './TODOListViewer';
+import ControlPanel from './ControlPanel';
+import {useSelector} from "react-redux";
+
+const TODOListMainView = () => {
+    const todoList = useSelector(state => state.todoReducer.TODOList);
+    const [selectedCategory, setSelectedCategory] = useState('');
+
+    const getVisibleTODOs = () => {
+        if (selectedCategory === '') {
+            return todoList;
+        }
+        return todoList.filter((todo) => todo.category === selectedCategory);
+    };
+
+    let visibleTODOList = getVisibleTODOs();
```

Comment:
you do not need to set a variable here, just call `getVisibleTODOs` where needed

Reply:
Ok. Advice taken. Thanks!